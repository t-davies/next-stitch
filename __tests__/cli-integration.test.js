const { system, filesystem } = require('gluegun');

const cwd = filesystem.path(__dirname, '..');

const cli = app => async cmd =>
  system.run('node ' + filesystem.path(cwd, 'bin', 'next-stitch') + ` ${cmd}`, {
    cwd: filesystem.path(cwd, app)
  });

test('prepare pages', async () => {
  const output = await cli('example')('prepare pages');
  expect(output).toContain('pages bundle created');

  expect(
    filesystem.find('./example/dist', { matching: '**/*' })
  ).toMatchSnapshot();

  filesystem.remove('./example/dist');
});

test('stitch', async () => {
  const output = await cli('example-shell')('stitch');
  expect(output).toContain('stitched "@next-stitch/next-stitch-test-package"');

  expect(
    filesystem.find('./example-shell/src/pages', { matching: '**/*' })
  ).toMatchSnapshot();

  expect(
    filesystem.read('./example-shell/src/pages/about.ts')
  ).toMatchSnapshot();

  filesystem.remove('./example-shell/pages/nested');
  filesystem.remove('./example-shell/pages/about.ts');
});
