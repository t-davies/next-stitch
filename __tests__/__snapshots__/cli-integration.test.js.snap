// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prepare pages 1`] = `
Array [
  "example\\\\dist\\\\bundle-22e3f067.css",
  "example\\\\dist\\\\index-d1fe0eb3.js",
  "example\\\\dist\\\\index-d1fe0eb3.js.map",
  "example\\\\dist\\\\index-e089d697.js",
  "example\\\\dist\\\\index-e089d697.js.map",
  "example\\\\dist\\\\pages\\\\about.cjs",
  "example\\\\dist\\\\pages\\\\about.cjs.map",
  "example\\\\dist\\\\pages\\\\about.mjs",
  "example\\\\dist\\\\pages\\\\about.mjs.map",
  "example\\\\dist\\\\pages\\\\nested\\\\[param].cjs",
  "example\\\\dist\\\\pages\\\\nested\\\\[param].cjs.map",
  "example\\\\dist\\\\pages\\\\nested\\\\[param].mjs",
  "example\\\\dist\\\\pages\\\\nested\\\\[param].mjs.map",
]
`;

exports[`stitch 1`] = `
Array [
  "example-shell\\\\src\\\\pages\\\\about.ts",
  "example-shell\\\\src\\\\pages\\\\nested\\\\[param].ts",
  "example-shell\\\\src\\\\pages\\\\_app.tsx",
]
`;

exports[`stitch 2`] = `
"/* @ts-ignore */
export { default, getStaticProps } from '@next-stitch/next-stitch-test-package/dist/pages/about';
"
`;
